### PR TITLE
Setup/Add integration/deployment tests via pytest

### PR DESCRIPTION
**Pull request by aktech**
## Changes:

Sets up a simple framework for writing integration/deployment tests for the deployed QHub on minikube.
Currently added basic tests for Dask Gateway.

## Types of changes

What types of changes does your code introduce?

_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

Added tests for running via pytest.

### Requires testing

- [x] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [x] Yes
- [ ] No

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered and more.

### CODE REVIEW

File: environment-dev.yaml [hunk 1]
```diff
@@ -16,3 +16,7 @@ dependencies:
   - pytest
   - diagrams
   - jhub-client
+
+  # deployment/integration tests dependencies
+  - dask-gateway
+  - python-kubernetes
```
Comment (danlester) [inline environment-dev.yaml:22]: I think this should be `kubernetes` not `python-kubernetes` (which is very old, and doesn't work with the tests).

But also, note that `pip install -e .[dev]` does not make use of this environment-dev.yaml file, so really "kubernetes" and "dask-gateway" need to be added to the dev list in setup.py

**Review by danlester — CHANGES_REQUESTED** [review]
By the way, I still think it would be worthwhile adding a Dask-based Jupyter notebook to be run during the Cypress stage :) But maybe another time...

File: environment-dev.yaml (hunk 1 above)
Comment (aktech) [reply environment-dev.yaml:22]: The correct pip package is `kubernetes`, but the conda's `kubernetes` package didn't work for me yesterday and `python-kubernetes` seems to work, either ways I'll update the dev dependency.

**Review by danlester — APPROVED** [review]
Oh yes, I wasn't thinking about Conda. Either way, the new setup.py looks better for pip.

**Pull request merged**